kube/src/client/discovery.rs

```diff
@@ -0,0 +1,341 @@
+use crate::{api::GroupVersionKind, Client};
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::{APIResource, APIResourceList};
+use std::{cmp::Reverse, collections::HashMap};
+
+struct GroupVersionData {
+    version: String,
+    list: APIResourceList,
+    resources: Vec<APIResource>,
+}
```

Comment:
Does this struct map onto something particular in the go codebase?

Reply:
I guess client-go just uses `APIResource` and `APIResourceList` everywhere, but they don't feel rusty imho.